modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/MultiSelectTag.java

```diff
@@ -0,0 +1,337 @@
+	 *             #setLocator(Object)}
+	 */
+	@Deprecated
+	public void setValueLocator() {
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_clearAllTitle = "";
+		_closeButtonAriaLabel = "";
+		_disabled = false;
+		_disabledClearAll = false;
+		_helpText = "";
+		_inputName = "";
+		_inputValue = "";
+		_label = "";
+		_locator = null;
+		_selectedItems = new ArrayList<>();
+		_sourceItems = new ArrayList<>();
+		_valid = true;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/MultiSelect";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("clearAllTitle", _clearAllTitle);
+		props.put("closeButtonAriaLabel", _closeButtonAriaLabel);
+		props.put("disabled", _disabled);
+		props.put("disabledClearAll", _disabledClearAll);
+		props.put("helpText", _helpText);
+		props.put("inputName", _inputName);
+		props.put("inputValue", _inputValue);
+		props.put("isValid", _valid);
+		props.put("label", _label);
+
+		if (_locator != null) {
+			props.put("locator", _locator);
+		}
+
+		props.put("selectedItems", _selectedItems);
+		props.put("sourceItems", _sourceItems);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("form-group");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		String id = getId();
+
+		if (_label != null) {
+			jspWriter.write("<label");
+
+			if (Validator.isNotNull(id)) {
+				jspWriter.write(" for=\"");
+				jspWriter.write(id);
+				jspWriter.write("\"");
+			}
+
+			jspWriter.write(">");
+
+			jspWriter.write(_label);
+			jspWriter.write("</label>");
+		}
+
+		jspWriter.write("<div class=\"input-group\"><div class=\"");
+		jspWriter.write("input-group-item\"><div class=\"form-control ");
+		jspWriter.write("form-control-tag-group input-group\"><div class=\"");
+		jspWriter.write("input-group-item\">");
+
+		List<Map<String, String>> selectedItems = getSelectedItems();
+
+		if (!ListUtil.isEmpty(selectedItems)) {
+			for (Map<String, String> selectedItem : selectedItems) {
+				String selectedItemLabel = selectedItem.get("label");
+
+				LabelTag labelTag = new LabelTag();
+
+				labelTag.setDismissible(true);
+				labelTag.setLabel(selectedItemLabel);
+
+				labelTag.doTag(pageContext);
+
+				jspWriter.write("<input type=\"hidden\" name=\"");
+				jspWriter.write(selectedItemLabel);
+				jspWriter.write("\" ");
+				jspWriter.write("value=\"");
+				jspWriter.write(selectedItem.get("value"));
+				jspWriter.write("\" />");
+			}
+		}
+
+		jspWriter.write("<input class=\"form-control-inset\"");
+
+		if (_disabled) {
+			jspWriter.write(" disabled");
+		}
+
+		if (Validator.isNotNull(id)) {
+			super.writeIdAttribute();
+		}
+
+		jspWriter.write(" type=\"text\" /></div>");
```

Comment:
Missing input `value` attribute (`inputValue`)

Reply:
@kresimir-coko changed that [here](https://github.com/liferay-frontend/liferay-portal/pull/528/commits/6657e5f4bb0a52f316d0672f5065f3f225de801b)